File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task188.java
import java.util.concurrent.Semaphore;

public class Task188 {

    // Runs the FooBar threading task and returns the concatenated result
    public static String runFooBar(int n) {
        if (n < 1 || n > 1000) {
            return "";
        }
        FooBar fb = new FooBar(n);
        Thread t1 = new Thread(() -> {
            try {
                fb.foo();
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
        });
        Thread t2 = new Thread(() -> {
            try {
                fb.bar();
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
        });

        t1.start();
        t2.start();
        try {
            t1.join();
            t2.join();
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
        return fb.getOutput();
    }

    // Inner class implementing the FooBar logic using semaphores
    static final class FooBar {
        private final int n;
        private final Semaphore fooSem = new Semaphore(1);
        private final Semaphore barSem = new Semaphore(0);
        private final StringBuilder sb = new StringBuilder();

        FooBar(int n) {
            this.n = n;
        }

        public void foo() throws InterruptedException {
            for (int i = 0; i < n; i++) {
                fooSem.acquire();
                sb.append("foo");
                barSem.release();
            }
        }

        public void bar() throws InterruptedException {
            for (int i = 0; i < n; i++) {
                barSem.acquire();
                sb.append("bar");
                fooSem.release();
            }
        }

        public String getOutput() {
            return sb.toString();
        }
    }

    public static void main(String[] args) {
        int[] tests = {1, 2, 3, 5, 10};
        for (int n : tests) {
            String out = runFooBar(n);
            System.out.println(out);
        }
    }
}

File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task188.py
import threading

class FooBar:
    def __init__(self, n: int):
        self.n = n
        self.foo_sem = threading.Semaphore(1)
        self.bar_sem = threading.Semaphore(0)
        self.out = []

    def foo(self):
        for _ in range(self.n):
            self.foo_sem.acquire()
            try:
                self.out.append("foo")
            finally:
                self.bar_sem.release()

    def bar(self):
        for _ in range(self.n):
            self.bar_sem.acquire()
            try:
                self.out.append("bar")
            finally:
                self.foo_sem.release()

def run_foobar(n: int) -> str:
    if n < 1 or n > 1000:
        return ""
    fb = FooBar(n)
    t1 = threading.Thread(target=fb.foo, daemon=True)
    t2 = threading.Thread(target=fb.bar, daemon=True)
    t1.start()
    t2.start()
    t1.join()
    t2.join()
    return "".join(fb.out)

if __name__ == "__main__":
    tests = [1, 2, 3, 5, 10]
    for n in tests:
        print(run_foobar(n))